src/Hpack/Util.hs

```diff
@@ -122,13 +122,24 @@ expandGlobs name dir patterns = do
     fromResult :: GlobResult -> ([String], [FilePath])
     fromResult (GlobResult pattern compiledPattern files) = case files of
       [] -> (warning, literalFile)
-      xs -> ([], map normalize xs)
+      xs -> ([], map (quoteSpecial . normalize) xs)
```

Comment:
I think quoting is a representational concern.  As such, I would tend to do this at the point we actually render the `.cabal` file.  Does this make sense?

What we do want to do here though is `newtype`-wrap `FilePath` so that the type checker helps us to ensure that we actually do the quoting at all relevant places.

Reply:
Should I add a "quotable" string constructor to `Hpack.Render.Dsl.Value`? Or should I quote them before then?

If I take the first approach (which I like more), then the issue becomes:
`LineSeperatedList` must also be able to contain "quotable" strings. Which will cause a lot of unneeded wrapping/unwrapping.

But if I quote them in `Hpack.Render` and pass them in like they are now, that would defeat the whole point, at least the way I understand it.

I chose to try out the second approach. You can see how it looks in this pr: https://github.com/googleson78/hpack/pull/1/files

It seems to be working(TM), the tests don't compile, but I just wanted to see what you think first.